src/plugins/ExtendedProfileSampling.cpp

```diff
@@ -0,0 +1,862 @@
+                    double p_y {0.0};
+                    double p_z {0.0};
+                    double jEF_x {0.0};
+                    double jEF_y {0.0};
+                    double jEF_z {0.0};
+                    double numSamples {0.0};
+                    if (_countSamples.at(i) > 0ul) {
+                        numMolsPerStep = static_cast<double>(_numMolecules_accum.at(i))/_countSamples.at(i);
+                        rho         = _density_accum.at(i)           /_countSamples.at(i);
+                        T           = _temperature_accum.at(i)       /_countSamples.at(i);
+                        ekin        = _ekin_accum.at(i)              /_countSamples.at(i);
+                        epot        = _epot_accum.at(i)              /_countSamples.at(i);
+                        p           = _pressure_accum.at(i)          /_countSamples.at(i);
+                        chemPot_res = _chemPot_accum.at(i)           /_countSamples.at(i);
+                        numTest     = static_cast<double>(_countNTest_accum.at(i))/_countSamples.at(i);
+                        T_x         = _temperatureVect_accum[0].at(i)/_countSamples.at(i);
+                        T_y         = _temperatureVect_accum[1].at(i)/_countSamples.at(i);
+                        T_z         = _temperatureVect_accum[2].at(i)/_countSamples.at(i);
+                        v_x         = _velocityVect_accum[0].at(i)   /_countSamples.at(i);
+                        v_y         = _velocityVect_accum[1].at(i)   /_countSamples.at(i);
+                        v_z         = _velocityVect_accum[2].at(i)   /_countSamples.at(i);
+                        p_x         = _pressureVect_accum[0].at(i)   /_countSamples.at(i);
+                        p_y         = _pressureVect_accum[1].at(i)   /_countSamples.at(i);
+                        p_z         = _pressureVect_accum[2].at(i)   /_countSamples.at(i);
+                        jEF_x       = _energyfluxVect_accum[0].at(i) /_countSamples.at(i);
+                        jEF_y       = _energyfluxVect_accum[1].at(i) /_countSamples.at(i);
+                        jEF_z       = _energyfluxVect_accum[2].at(i) /_countSamples.at(i);
+                        numSamples  = _countSamples.at(i);
+                    }
+                    if ((_chemPot_accum.at(i) > 0.0) and (_countNTest_accum.at(i) > 0ul) and (_countSamples.at(i) > 0ul)) {
+                        numTest     = _countNTest_accum.at(i)/_countSamples.at(i);
+                        chemPot_res = -log(_chemPot_accum.at(i)/_countNTest_accum.at(i)) + log(rho);
+                    }
+                    ofs << std::setw(24) << std::scientific << std::setprecision(std::numeric_limits<double>::digits10) << numMolsPerStep
+                        << std::setw(24) << std::scientific << std::setprecision(std::numeric_limits<double>::digits10) << rho
+                        << std::setw(24) << std::scientific << std::setprecision(std::numeric_limits<double>::digits10) << T
```

Comment:
I would advise setting this width via (`constexpr`) variable so it is easier to change if necessary.